src/modules/fancyzones/editor/FancyZonesEditor/MyButton.xaml.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Windows;
+using System.Windows.Automation;
+using System.Windows.Automation.Peers;
+using System.Windows.Automation.Provider;
+using System.Windows.Controls;
+
+namespace FancyZonesEditor
+{
+    public partial class MyButton : Button
+    {
+        public MyButton()
+            : base()
+        {
+            InitializeComponent();
+            Click += OnClick;
+        }
+
+        public string Value
+        {
+            get { return (string)GetValue(ValueProperty); }
+            set { SetValue(ValueProperty, value); }
+        }
+
+        public static readonly DependencyProperty ValueProperty =
+        DependencyProperty.Register(
+            "Value", typeof(string), typeof(MyButton));
+
+        private void OnClick(object sender, RoutedEventArgs e)
+        {
+            if (AutomationPeer.ListenerExists(AutomationEvents.PropertyChanged))
+            {
+                MyButtonAutomationPeer peer =
+                    UIElementAutomationPeer.FromElement(this) as MyButtonAutomationPeer;
+
+                if (peer != null)
+                {
+                    peer.RaisePropertyChangedEvent(
+                        ValuePatternIdentifiers.ValueProperty,
+                        null,
+                        Value);
```

Comment:
It might be a better implementation to -- perhaps not use a custom button class and -- use [UIElementAutomationPeer's **RaiseNotificationEvent** method](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer.raisenotificationevent?view=winrt-20348). Terminal uses this to announce that a search yielded no results [here](https://github.com/microsoft/terminal/blob/59166faa27d4b692d97a0314b4e32406b205d9a0/src/cascadia/TerminalApp/CommandPalette.cpp#L554-L558).

If you already have a codebehind for CanvasEditorWindow, or something that responds to the Click event and still has access to the UI tree, you can greatly simplify the implementation.

In Terminal's case, we use the _text input field_'s automation peer since that was the element the user was interacting with.

Reply:
Er, this is WPF. Hmm. Sec.

Reply:
It looks like the UIA API that the UWP implementation is based on came from Windows 10 Fall Creator Update ... which suggests that there is 0 change there is a WPF equivalent. That's quite frustrating. I'm going to clear my block, as I don't have any better suggestions.